shell/common/rasterizer.cc

```diff
@@ -258,11 +258,19 @@ sk_sp<SkImage> Rasterizer::DoMakeRasterSnapshot(
   sk_sp<SkImage> result;
   SkImageInfo image_info = SkImageInfo::MakeN32Premul(
       size.width(), size.height(), SkColorSpace::MakeSRGB());
-  if (surface_ == nullptr || surface_->GetContext() == nullptr) {
+
+  std::unique_ptr<Surface> pbuffer_surface;
+  if (!surface_) {
+    pbuffer_surface = delegate_.CreateSnapshotSurface();
+  }
+
```

Comment:
I'd ahve to manually delete the surface if it came from CreateSnapshotSurface then though. like `pbuffer_surface.release()` and then a `delete` at the end if it's not nullptr and not the `surface_.get()`. Is there some way I'm missing?

Reply:
`pbuffer_surface` would be a `unique_ptr` declared at the same scope as `snapshot_surface`.  If `pbuffer_surface` is assigned then it will be deleted when the function exits.

Reply:
Ahh I get it now.

Reply:
Done